yarn-project/cli-wallet/src/cmds/create_account.ts

```diff
@@ -72,7 +73,12 @@ export async function createAccount(
       ...(await feeOpts.toDeployAccountOpts(wallet)),
     };
     if (feeOpts.estimateOnly) {
-      const gas = await account.estimateDeploymentGas(deployOpts);
+      const fee =
+        !deployOpts?.deployWallet && deployOpts?.fee
+          ? { ...deployOpts.fee, paymentMethod: await account.getSelfPaymentMethod(deployOpts.fee.paymentMethod) }
+          : deployOpts?.fee;
+      const deployMethod = await account.getDeployMethod(deployOpts.deployWallet);
+      const gas = await deployMethod.estimateGas({ ...deployOpts, fee, universalDeploy: true });
```

Comment:
I know this is a wip with comments and polishing missing, but this bit is just too cryptic to follow atm

Reply:
Added comment